tools/CheckDates.scala

```diff
@@ -0,0 +1,110 @@
+import sys.process._
+
+object CheckDates {
+  val usage = """
+
+    This tool compares a copyright date in files with the date of last change in git.
+    If the message of last change contains "Documents date fixed" it will be ignored and 
+    we will take next commit.
+
+    analyzeDates - just shows if anything is ok
+    fixWrongDates - actual fixing
+
+    From root folder:
+    Usage: scala tools/CheckDates.scala analyzeDates
+           scala tools/CheckDates.scala fixWrongDates
+
+  """
+
+  def readFilesList(): Array[String] = {
+    val myval =  ("find . -name *.scala ! -path ./tools/*  -type f") !!
+    val files_array = myval.split("\n").map(_.toString).distinct
+    
+    return files_array
+  }
+
+  def readFile(pathToFile: String): String = {
+    val source = scala.io.Source.fromFile(pathToFile)
+    val lines = try source.mkString finally source.close()
+
+    return lines
+  }
+
+  def githubDate(pathToFile: String): String = {
+    val buf_dates = (s"git log -3  --format=%cd $pathToFile" !!)
+    val buf_messages = (s"git log -3  $pathToFile" !!)
+    val messages = buf_messages.split("commit").distinct.drop(1)
+    val dates = buf_dates.split("\\n").distinct
+    var gits_info = Array[(String, String)]()
+    for (a <- 0 to dates.length - 1) {
+      gits_info = gits_info :+ (messages(a),dates(a))
+    }
+    val res = gits_info.filter(!_._1.contains("Documents date fixed"))
```

Comment:
Deploying a case class will allow the _n syntax to be dropped resulting in more maintainable code.